src/Runner.Worker/ExecutionContext.cs

```diff
@@ -924,6 +931,47 @@ public void PublishStepTelemetry()
                 // Add to the global steps telemetry only if we have something to log.
                 if (!string.IsNullOrEmpty(StepTelemetry?.Type))
                 {
+                    if (!IsEmbedded)
+                    {
+                        StepTelemetry.Result = _record.Result;
```

Comment:
only set for not embedded step, since the whole composite action will have a single result.